search/src/Components/upload/react-images-upload/index.js

```diff
@@ -85,55 +87,47 @@ class ReactImageUploadComponent extends React.Component {
           type: ERROR.FILESIZE_TOO_LARGE
         });
         fileErrors.push(fileError);
+        this.setState({fileErrors});
         continue;
       }
-
-      allFilePromises.push(this.readFile(file));
-    }
-
-    this.setState({
-      fileErrors
-    });
-
-    const {singleImage} = this.props;
-
-    Promise.all(allFilePromises).then(newFilesData => {
-      const dataURLs = singleImage?[]:this.state.pictures.slice();
-      const files = singleImage?[]:this.state.files.slice();
-
-      newFilesData.forEach(newFileData => {
-        this.uploadFileToServer(dataURLs, files, newFileData);
+      // Check for duplication
+      if (!this.props.images.includes(file.name) || filesName.includes(file.name)){
+        fileError = Object.assign(fileError, {
+          type: ERROR.FILE_DUPLICATE
+        });
+        fileErrors.push(fileError);
+        this.setState({fileErrors});
+        continue;
+      }
+      this.readFile(file).then(newFileData => {
+        this.uploadFileToServer(newFileData);
       });
-    });
+    }
   }
 
   /*
      Customised function modified on original codebase
      Copyright (c) 2020 Zheng Li
    */
-  uploadFileToServer(dataURLs, files, newFileData) {
-    const filesName = files.map(file => file.name);
-    if (!this.props.images.includes(newFileData.file.name) || filesName.includes(newFileData.file.name)){
-        return
-    }
-    else {
-        const body = new FormData()
-        body.append('upload_image', newFileData.file)
-        body.append('upload_id', this.props.upload_id)
-        axios({
-            method: 'post',
-            url: this.props.uploadURL,
-            data: body,
-            headers: {'Content-Type': 'multipart/form-data', 'X-CSRFToken': Cookies.get('csrftoken') }
-        }).then(res => {
-            if(res.status === 200){
-                dataURLs.push(newFileData.dataURL);
-                files.push(newFileData.file);
-                this.setState({pictures: dataURLs, files: files});
-                this.props.handleUploaded(files.map(file => file.name));
-            }
-        })
-    }
+  async uploadFileToServer(newFileData) {
```

Comment:
Let's make this not async. Too confusing to read. Revert to using axios().then(...)